### DISCUSSION TITLE
Frequent REFRESH_TOKEN_ERROR

### DISCUSSION THREAD

**Discussion by giordanodavid** [keycloak/keycloak#21128] [Q&A]
Hello,

I have an application on which users are using Authorization Code to log in. So, an access token and a refresh token are created.
The first login is successful as you can see :
![Capture](https://github.com/keycloak/keycloak/assets/127959850/ef3d797f-84aa-4253-a66b-e7c072726563)

However, when the access token expires after 10 min, and a new one needs to be created thanks to the refresh token, my user got the following error :
`2023-06-21 13:24:51,650 WARN  [org.keycloak.events] (executor-thread-9754) type=REFRESH_TOKEN_ERROR, realmId=apim-ww-play, clientId=GfqOjCNoA1eKX5hKEMLZmowGjx4nFasR, userId=e7e83fda-aba3-4237-b647-cb20834a733c, ipAddress=10.68.189.165, error=invalid_token, grant_type=refresh_token, refresh_token_type=Refresh, refresh_token_id=58000844-1deb-4480-8187-cdfc243dc3b1, client_auth_method=client-secret`
I don't have more information except this log.

The refresh token used is this one :
`eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJlNWQ2YjRmZC0yOWNlLTQzM2MtYjZjYi01Yjg1ZjAwNDgyNjgifQ.eyJleHAiOjE2ODczNTY4NzgsImlhdCI6MTY4NzM1MzI3OCwianRpIjoiNTgwMDA4NDQtMWRlYi00NDgwLTgxODctY2RmYzI0M2RjM2IxIiwiaXNzIjoiaHR0cHM6Ly9rZXljbG9hay5yaWNoZW1vbnQuY29tL2F1dGgvcmVhbG1zL2FwaW0td3ctcHJvZCIsImF1ZCI6Imh0dHBzOi8va2V5Y2xvYWsucmljaGVtb250LmNvbS9hdXRoL3JlYWxtcy9hcGltLXd3LXByb2QiLCJzdWIiOiJlN2U4M2ZkYS1hYmEzLTQyMzctYjY0Ny1jYjIwODM0YTczM2MiLCJ0eXAiOiJSZWZyZXNoIiwiYXpwIjoiR2ZxT2pDTm9BMWVLWDVoS0VNTFptb3dHang0bkZhc1IiLCJzZXNzaW9uX3N0YXRlIjoiYTU3ZTNlNDMtOTNlYS00NDJhLWJkMzEtOTEwNjI2MjEwZDcxIiwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCIsInNpZCI6ImE1N2UzZTQzLTkzZWEtNDQyYS1iZDMxLTkxMDYyNjIxMGQ3MSJ9.EwhDt4J1-IG7nD228sosIFPpX3jbUtvDoZOwfptgn-E`
We can see that the token is not expired and that the subject = the userId of my user.

I don't understand why Keycloak says that the refresh token is invalid.

Your help would be appreciated to solve this issue.

Thanks in advance.

David

**Comment by sschu** [comment]
Can you replicate this problem somewhere in isolation and increase the log level of Keycloak so you get more information on what is happening?

**Reply by giordanodavid** [reply]
Hello,
Thanks for your answer. Even with a DEBUG trace level, I don't have more relevant informations, unfortunately...

**Reply by sschu** [reply]
Maybe increase to TRACE?

**Reply by ssonparo** [reply]
@giordanodavid: I am facing the same issue not even able to reproduce .
